pkg/api/v1beta3/dynakube/dynakube_types.go

```diff
@@ -0,0 +1,175 @@
+	Value string `json:"value,omitempty"`
+
+	// Secret containing proxy URL.
+	// +nullable
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Proxy secret",order=33,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:io.kubernetes:Secret"}
+	ValueFrom string `json:"valueFrom,omitempty"`
+}
+
+type DynaKubeValueSource struct { //nolint:revive
+	// Custom properties value.
+	// +nullable
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Custom properties value",order=32,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:com.tectonic.ui:text"}
+	Value string `json:"value,omitempty"`
+
+	// Custom properties secret.
+	// +nullable
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Custom properties secret",order=33,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:io.kubernetes:Secret"}
+	ValueFrom string `json:"valueFrom,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DynaKube is the Schema for the DynaKube API
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=dynakubes,scope=Namespaced,categories=dynatrace,shortName={dk,dks}
+// +kubebuilder:printcolumn:name="ApiUrl",type=string,JSONPath=`.spec.apiUrl`
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.phase`
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+// +operator-sdk:csv:customresourcedefinitions:displayName="Dynatrace DynaKube"
+// +operator-sdk:csv:customresourcedefinitions:resources={{StatefulSet,v1,},{DaemonSet,v1,},{Pod,v1,}}.
+type DynaKube struct {
+	metav1.TypeMeta `json:",inline"`
+
+	Status            DynaKubeStatus `json:"status,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec DynaKubeSpec `json:"spec,omitempty"`
+}
+
+// DynaKubeSpec defines the desired state of DynaKube
+// +k8s:openapi-gen=true
+type DynaKubeSpec struct { //nolint:revive
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// Set custom proxy settings either directly or from a secret with the field proxy.
+	// Note: Applies to Dynatrace Operator, ActiveGate, and OneAgents.
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Proxy",order=3,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced","urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	Proxy *DynaKubeProxy `json:"proxy,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	ExtensionExecutionController ExtensionExecutionControllerSpec `json:"extensionExecutionController,omitempty"`
```

Comment:
This element should be in a `templates` subsection, not directly on this level.

Reply:
fixed.